it/src/test/java/org/corfudb/universe/scenario/RotateLinkFailureIT.java

```diff
@@ -0,0 +1,141 @@
+
+public class RotateLinkFailureIT extends GenericIntegrationTest {
+
+    /**
+     * Test cluster behavior when rotating link failure among nodes
+     * <p>
+     * 1) Deploy and bootstrap a three nodes cluster
+     * 2) Create a link failure between node0 and node1
+     * 3) Create a link failure between node1 and node2
+     *    and heal previous link failure
+     * 4) Create a link failure between node2 and node0
+     *    and heal previous link failure
+     * 5) Reverse rotation direction, create a link failure
+     *    between node1 and node2 and heal previous link failure
+     * 6) Verify layout and data path after each rotation
+     * 7) Recover cluster by removing all link failures
+     * 8) Verify layout, cluster status and data path
+     */
+    @Test(timeout = 300000)
+    public void rotateLinkFailureTest() {
+        getScenario().describe((fixture, testCase) -> {
+            CorfuCluster corfuCluster = universe.getGroup(fixture.getCorfuCluster().getName());
+
+            CorfuClient corfuClient = corfuCluster.getLocalCorfuClient();
+
+            CorfuTable<String, String> table = corfuClient.createDefaultCorfuTable(DEFAULT_STREAM_NAME);
+            for (int i = 0; i < DEFAULT_TABLE_ITER; i++) {
+                table.put(String.valueOf(i), String.valueOf(i));
+            }
+
+            testCase.it("Should rotate link failures among cluster", data -> {
+                CorfuServer server0 = corfuCluster.getServerByIndex(0);
+                CorfuServer server1 = corfuCluster.getServerByIndex(1);
+                CorfuServer server2 = corfuCluster.getServerByIndex(2);
+
+                // 1st link failure rotation, disconnect between server0 and server1
+                server0.disconnect(Collections.singletonList(server1));
+                waitForUnresponsiveServersChange(size -> size == 1, corfuClient);
+
+                String firstServerToKick = Collections.max(Arrays.asList(server0.getEndpoint(), server1.getEndpoint()));
+                assertThat(corfuClient.getLayout().getUnresponsiveServers()).containsExactly(firstServerToKick);
+
+                Layout lastLayout = corfuClient.getLayout();
+
+                // Verify data path working fine
+                for (int i = 0; i < DEFAULT_TABLE_ITER; i++) {
+                    assertThat(table.get(String.valueOf(i))).isEqualTo(String.valueOf(i));
+                }
+
+                // 2nd link failure rotation, disconnect between server1 and server2
+                // and heal previous link failure between server0 and server1
+                server1.disconnect(Collections.singletonList(server2));
+                server0.reconnect(Collections.singletonList(server1));
+
+                // Wait for some time to ensure cluster stabilizes
+                // Server1 should stay in unresponsive set, no layout change
+                waitUninterruptibly(20);
```

Comment:
Maybe it's better to pass Duration instead of int?

Reply:
Done.